openreview/conference/invitation.py

```diff
@@ -516,3 +516,83 @@ def set_reviewer_recruiter_invitation(self, conference_id, options = {}):
+
+        # Create super invitation with a webfield
+        recommendation_invitation = openreview.Invitation(
+            id = conference.get_id() + '/-/Recommendation',
+            duedate = tools.datetime_millis(due_date),
+            readers = [conference.get_program_chairs_id(), conference.get_area_chairs_id()],
+            invitees = [],
+            writers = [conference.get_id()],
+            signatures = [conference.get_id()],
+            multiReply = True,
+            reply = {
+                'invitation': conference.get_blind_submission_id(),
+                'readers': {
+                    'description': 'The users who will be allowed to read the above content.',
+                    'values-copied': [conference.get_id(), '{signatures}']
+                },
+                'signatures': {
+                    'description': 'How your identity will be displayed with the above content.',
+                    'values-regex': '~.*'
+                },
+                'content': {
+                    'tag': {
+                        'description': 'Recommend a reviewer to review this paper',
+                        'order': 1,
+                        'required': True,
+                        'values-url': '/groups?id=' + conference.get_reviewers_id()
+                    }
+                }
+            }
+        )
+
+        recommendation_invitation = self.client.post_invitation(recommendation_invitation)
+        # Create subinvitation with different list of reviewers, bid, tpms score.
+
+        for note in notes_iterator:
+            reviewers = []
+            assignment_note = assignment_note_by_forum.get(note.id)
+            if assignment_note:
+                for group in assignment_note['assignedGroups']:
+                    reviewers.append('{profileId} (A) - Bid: {bid} - Tpms: {tpms}'.format(
+                        profileId = group.get('userId'),
+                        bid = group.get('scores').get('bid'),
+                        tpms = group.get('scores').get('affinity'))
+                    )
+                for group in assignment_note['alternateGroups']:
+                    reviewers.append('{profileId} - Bid: {bid} - Tpms: {tpms}'.format(
+                        profileId = group.get('userId'),
+                        bid = group.get('scores').get('bid'),
+                        tpms = group.get('scores').get('affinity'))
+                    )
+            else:
+                raise openreview.OpenReviewException('Assignment note not found for ' + note.id)
+            paper_recommendation_invitation = openreview.Invitation(
+                id = conference.get_id() + '/-/Paper{number}/Recommendation'.format(number = note.number),
+                super = recommendation_invitation.id,
+                invitees = [conference.get_program_chairs_id(), conference.get_id() + '/Paper{number}/Area_Chairs'.format(number = note.number)],
```

Comment:
If possible, I think we should avoid using 'Area_Chairs' here.

Reply:
As I talked with Michael a few days ago, we are doing the assignments using 'Area_Chairs' and 'AnonReviewers' strings, even in UAI. Otherwise I need to parametrize these strings everywhere in the builder and I decided to do this later.